webapp/src/DOMJudgeBundle/Controller/Jury/ConfigController.php

```diff
@@ -0,0 +1,118 @@
+    {
+        /** @var Configuration[] */
+        $options = $this->entityManager->getRepository('DOMJudgeBundle:Configuration')->findAll();
+        if ($request->getMethod() == 'POST' && $request->request->has('save')) {
+            foreach ($options as $option) {
+                if ($option->getType() == 'bool') {
+                    $val = $request->request->has('config_' . $option->getName());
+                    $option->setValue($val);
+                    continue;
+                }
+                if (!$request->request->has('config_' . $option->getName())) {
+                    continue;
+                }
+                if ($option->getType() == 'int' || $option->getType() == 'string') {
+                    $option->setValue($request->request->get('config_' . $option->getName()));
+                } else if ($option->getType() == 'array_val') {
+                    $vals = $request->request->get('config_' . $option->getName());
+                    $result = array();
+                    foreach ($vals as $data) {
+                        if (!empty($data)) {
+                            $result[] = $data;
+                        }
+                    }
+                    $option->setValue($result);
+                } else if ($option->getType() == 'array_keyval') {
+                    $vals = $request->request->get('config_' . $option->getName());
+                    $result = array();
+                    foreach ($vals as $data) {
+                        if (!empty($data['key'])) {
+                            $result[$data['key']] = $data['val'];
+                        }
+                    }
+                    $option->setValue($result);
+                }
+            }
+        }
+        $this->entityManager->flush();
+        /** @var Configuration[] */
+        $options = $this->entityManager->getRepository('DOMJudgeBundle:Configuration')->findAll();
+        $categories = array();
+        foreach ($options as $option) {
+            if (!in_array($option->getCategory(), $categories)) {
+                $categories[] = $option->getCategory();
+            }
+        }
+        $all_data = array();
+        foreach ($categories as $category) {
+            $data = array();
+            foreach ($options as $option) {
+                if ($option->getCategory() !== $category) {
+                    continue;
+                }
+                $data[] = array(
+                    'name' => $option->getName(),
+                    'type' => $option->getType(),
+                    'value' => $option->getValue(),
+                    'description' => $option->getDescription()
+                );
+            }
+            $all_data[] = array(
+                'name' => $category,
+                'data' => $data
+            );
+        }
+        return $this->render('@DOMJudge/jury/config.html.twig', [
+            'options' => $all_data,
```

Comment:
If you set a value like `edited` to true and pass it to Twig when settings have been saved, you can do something like this in the twig:

```twig
    {% if edited %}
        {% include '@DOMJudge/jury/refresh_scoreboard.html.twig' with {message: 'After changing specific settings, you might need to refresh the scoreboard.'} %}
    {% endif %}
```

Reply:
Done.